backbone.js

```diff
@@ -1202,6 +1217,61 @@
 
   });
 
+  // Defining an @@iterator method implements JavaScript's Iterable protocol.
+  // In modern ES2015 browsers, this value is found at Symbol.iterator, however
+  // older browsers do not define Symbol and substitute the string '@@iterator'.
+  var $$iterator = typeof Symbol === 'function' && Symbol.iterator || '@@iterator'; // eslint-disable-line no-undef
+  Collection.prototype[$$iterator] = Collection.prototype.values;
+
+  // CollectionIterator
+  // ------------------
+
+  // A CollectionIterator implements JavaScript's Iterator protocol, allowing the
+  // use of `for of` loops in modern browsers and interoperation between
+  // Backbone.Collection and other JavaScript functions and third-party libraries
+  // which can operate on Iterables.
+  var CollectionIterator = function(collection, kind) {
+    this._collection = collection;
+    this._kind = kind;
+    this._index = 0;
+  };
+
+  // All Iterators should themselves be Iterable.
+  CollectionIterator.prototype[$$iterator] = function() {
+    return this;
+  };
+
+  CollectionIterator.prototype.next = function() {
+    if (this._collection) {
+
+      // Only continue iterating if the iterated collection is long enough.
+      if (this._index < this._collection.length) {
+        var model = this._collection.at(this._index);
+        this._index++;
+
+        // Construct a value depending on what kind of values should be iterated.
+        var value;
+        if (this._kind === 'value') {
```

Comment:
Do you have a preferred enum style in mind? I don't feel strongly about this, happy to take suggestions.